docs/API/V1/upload.md

```diff
@@ -0,0 +1,101 @@
+# `PUT /api/v1/upload/<file>`
+
+This API creates a dataset resource by uploading a performance tarball.
+
+Primarily this is expected to be a native Pbench Agent tarball with a specific
+structure; however with the `server.archiveonly` metadata key the Pbench Server
+can be used to archive and manage metadata for any tarball.
+
+## URI parameters
+
+`<file>` string \
+The initial name of the dataset; if `server.archiveonly` is not set, the name must
+match the internal name recorded by the Pbench Agent.
+
+## Query parameters
+
+`access` [ `private` | `public` ] \
+The desired initial access scope of the dataset. Select `public` to make the dataset
+accessible to all clients, or `private` to make the dataset accessible only
+to the owner. The default is `private`.
+
+For example, `?access=public`
+
+`metadata` metadata keys \
+A valid Pbench Server username to be given ownership of the specified dataset.
+This requires the authenticated user to hold `ADMIN` role, essentially granting
+full access to both the current and new owners.
+
+In particular the `server.archiveonly` metadata key allows telling the Pbench
+Server that the tarball should not be unpacked, analyzed, or indexed, for example
+when it doesn't have the expected Pbench Agent `metadata.log`. The tarball will be
```

Comment:
Replace "have" with something like "contain" or "include".  And, I would replace "it" with "the tarball".